core/src/test/java/org/jboss/modcluster/load/impl/DynamicLoadBalanceFactorProviderTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2017, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.modcluster.load.impl;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jboss.modcluster.container.Engine;
+import org.jboss.modcluster.load.metric.NodeUnavailableException;
+import org.jboss.modcluster.load.metric.LoadMetric;
+import org.jboss.modcluster.load.metric.impl.AbstractLoadMetric;
+import org.junit.Test;
+
+/**
+ * @author Radoslav Husar
+ */
+public class DynamicLoadBalanceFactorProviderTest {
+
+    @Test
+    public void getLoadBalanceFactor() throws Exception {
+        Set<LoadMetric> metrics = new HashSet<>();
+        metrics.add(new NodeUnavailableExceptionThrowingLoadMetric());
+        DynamicLoadBalanceFactorProvider provider = new DynamicLoadBalanceFactorProvider(metrics);
```

Comment:
I could do this for metric (if that was the intention?) but I don't think that's overly practical.